internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -160,6 +166,69 @@ Assemble(const std::deque<BitsetType>& srcs) -> BitsetType {
     return res;
 }
 
+void
+AppendOneChunk(BitsetType& result, const FixedVector<bool>& chunk_res) {
+    // Append a value once instead of BITSET_BLOCK_BIT_SIZE times.
+    auto AppendBlock = [&result](const bool* ptr, int n) {
+        for (int i = 0; i < n; ++i) {
+            BitSetBlockType val = 0;
+            // This can use CPU SIMD optimzation
+            uint8_t vals[BITSET_BLOCK_SIZE] = {0};
+            for (size_t j = 0; j < 8; ++j) {
+                for (size_t k = 0; k < BITSET_BLOCK_SIZE; ++k) {
+                    vals[k] |= uint8_t(*(ptr + k * 8 + j)) << j;
```

Comment:
if this code is performance leak, please write avx/sse/neon intrinsic functions.

Reply:
A good ideal but now performace is quiet good because compiler has optimized it partly

Reply:
have implement it and use these functions next pr